console/src/components/CompactGPS/CompactGPS.js

```diff
@@ -0,0 +1,212 @@
+import React from 'react';
+import mapboxgl from 'mapbox-gl';
+console.log(process.env.REACT_APP_MAPBOX)// eslint-disable-line
+
+mapboxgl.accessToken = process.env.REACT_APP_MAPBOX; // eslint-disable-line
+
+const getRoute = async (start, end) => {
+	const response = await window.fetch('https://api.mapbox.com/directions/v5/mapbox/driving/' + start[0] + ',' + start[1] + ';' + end[0] + ',' + end[1] + '?geometries=geojson&access_token=' + mapboxgl.accessToken);
+	return await response.json();
+};
+
+const markerLayer = {
+	'id': 'symbols',
+	'type': 'symbol',
+	'source': {
+		'type': 'geojson',
+		'data': {
+			'type': 'FeatureCollection',
+			'features': [
+				{
+					'type': 'Feature',
+					'properties': {
+						'description': 'Tasty Subs & Pizza'
+					},
+					'geometry': {
+						'type': 'Point',
+						'coordinates': [
+							-121.995479, 37.384965
+						]
+					}
+				},
+				{
+					'type': 'Feature',
+					'properties': {
+						'description': 'Hobee\'s'
+					},
+					'geometry': {
+						'type': 'Point',
+						'coordinates': [
+							-122.023559, 37.396325
+						]
+					}
+				},
+				{
+					'type': 'Feature',
+					'properties': {
+						'description': 'Dishdash'
+					},
+					'geometry': {
+						'type': 'Point',
+						'coordinates': [
+							-122.026600, 37.376994
+						]
+					}
+				}
+			]
+		}
+	},
+	'layout': {
+		'icon-image': 'rocket-15'
+	}
+};
+
+class CompactGps extends React.Component {
+	constructor (props) {
+		super(props);
+		if (this.props.theme === 'carbon') {
+			this.style = 'mapbox://styles/haileyr/cjn4x0ynt04jq2qpf5sb21jc5';
+		} else if (this.props.theme === 'titanium') {
+			this.style = 'mapbox://styles/mapbox/dark-v9';
+		}
+	}
+
+
+
+	componentDidMount () {
+		let start = [-121.979125, 37.405189];
+		this.map = new mapboxgl.Map({
+			container: 'map',
+			style: this.style,
+			center: [-121.979125, 37.405189],
+			zoom: 12
+		});
+		this.map.addControl(new mapboxgl.GeolocateControl({
+			positionOptions: {
+				enableHighAccuracy: true
+			},
+			trackUserLocation: true
+		}));
+		this.map.on('load', () => {
+			this.map.addLayer(markerLayer);
+		});
+		this.map.on('click', 'symbols', (e) => {
+			let coordinates = e.features[0].geometry.coordinates.slice();
+			let description = e.features[0].properties.description;
+
+			while (Math.abs(e.lngLat.lng - coordinates[0]) > 180) {
+				coordinates[0] += e.lngLat.lng > coordinates[0] ? 360 : -360;
+			}
+
+			this.showPopup(coordinates, description);
+			this.drawDirection(start, coordinates);
+			this.map.flyTo({center: [(coordinates[0] + start[0]) / 2, (coordinates[1] + start[1]) / 2]});
+		});
+	}
+
+	componentWillUpdate () {
```

Comment:
This seems a little buggy.  Sometimes I get a dark map with `titanium` and when I switch to `carbon` it stays dark.  Sometimes the opposite occurs.  I'm not sure why, since the code looks fine, maybe it's just me :D

Reply:
This will be important to look into. Perhaps we need direct integration with the Skinnable hoc or the app's settings, rather than passing props down for the skin.